### ISSUE TITLE
v13 types issue

### ISSUE THREAD

**Issue by Techn1x** [ember-cli/ember-resolver#981]
FYI, after updating to v13, seems that types aren't seen correctly - wondering if something got misconfigured.

![Screenshot 2024-09-12 at 12 06 40 pm](https://github.com/user-attachments/assets/5aad94f4-693c-49a8-bbf0-21b2a98bce02)

Was able to work around this by adding this to my tsconfig

```ts
"paths": {
   ...
   "ember-resolver": ["node_modules/ember-resolver"]
},
```

**Comment by NullVoxPopuli** [comment]
thanks for your bug report!

does this change fix your issue? https://github.com/ember-cli/ember-resolver/pull/982

**Comment by ef4** [comment]
I can believe that the addition of `exports` in package.json changed typescript's resolution behavior. But I'm not able to reproduce in the app I checked. What is your `moduleResolution` setting in tsconfig? I'm guess this only breaks under some settings and not others.

Also yes, please confirm if NullVoxPopuli's fix above solves your problem, if so I will merge and release that.

**Comment by Techn1x** [comment]
Oops! My phone didn't seem to notify me of emails for a few days. Logged in and saw all my github notifications 😅 

I'll try this out first thing Monday (tomorrow for me) - I suspect it will work

**Comment by Techn1x** [comment]
Can confirm, that works great @NullVoxPopuli 

I removed the tsconfig workaround I added, restarted Glint to ensure type error showed again, edited the node_modules/ember-resolver/package.json file with the exports from that PR, restarted Glint, and the type is correctly found and works.

**Comment by Techn1x** [comment]
> What is your moduleResolution setting in tsconfig?

I don't set one actually, just extend from `@tsconfig/ember/tsconfig.json`

My tsconfig looks like this

```
{
  "extends": "@tsconfig/ember/tsconfig.json",
  "compilerOptions": {
    "declaration": false,
    "experimentalDecorators": true,
    "allowJs": true,
    "skipLibCheck": true,
    "baseUrl": ".",
    "noEmitOnError": false,
    "allowSyntheticDefaultImports": true,
    "noUncheckedIndexedAccess": false,
    "noPropertyAccessFromIndexSignature": false,
    "paths": {
      "district-ui-client/tests/*": ["tests/*"],
      "district-ui-client/*": ["app/*"],
      "*": ["types/*"],
      "ember-cli-flash/*": ["node_modules/ember-cli-flash"],
    }
  },
  "glint": {
    "environment": ["ember-loose", "ember-template-imports"]
  },
  "include": ["app/**/*", "tests/**/*", "types/**/*"]
}
```

**Comment by ijlee2** [comment]
I also ran into the error (app's `tsconfig.json` makes use of `@tsconfig/ember`):

```sh
[lint:types] app/app.ts:5:22 - error TS2307: Cannot find module 'ember-resolver' or its corresponding type declarations.
[lint:types] 
[lint:types] 5 import Resolver from 'ember-resolver';
[lint:types]                        ~~~~~~~~~~~~~~~~
[lint:types] 
[lint:types] tests/helpers/index.ts:6:27 - error TS2307: Cannot find module 'ember-resolver' or its corresponding type declarations.
[lint:types] 
[lint:types] 6 import type Resolver from 'ember-resolver';
[lint:types]                             ~~~~~~~~~~~~~~~~
[lint:types] 
```

I expect that #982 will fix it. 💯

**Comment by NullVoxPopuli** [comment]
Should be released in 13.0.1 :tada: 

lemme know if anything is borked
